### PR TITLE
Fix formatting and modes

### PR DESCRIPTION
**Pull request by iranzo**
Fixes formatting to markdown and shfmt on bash files + sets proper nodes for executables

Additionally adds .precommit configuration and flake8 formatting adapted to the precommit configuration used for the formatting:

IRT pre-commit: https://pre-commit.com/

### CODE REVIEW

**Comment by ptrnull** [comment]
/lgtm
Tested in my cluster, most of the changes are non required spaces or tabs.

**Review by ptrnull — APPROVED** [review]

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+        name: shfmt
+        minimum_pre_commit_version: 2.4.0
+        language: golang
+        additional_dependencies:
+        - mvdan.cc/sh/v3/cmd/shfmt@v3.1.1
+        entry: shfmt
+        args:
+        - -w
+        - -i
+        - '4'
```
Comment (dcritch) [inline .pre-commit-config.yaml:56]: nice work! though I'm a 2 space fan myself :)

**Review by dcritch — APPROVED** [review]

File: .pre-commit-config.yaml (hunk 1 above)
Comment (iranzo) [reply .pre-commit-config.yaml:56]: Do you want me to change it to '2' then?

Comment (iranzo) [reply .pre-commit-config.yaml:56]: or in the next pr ?

Comment (dcritch) [reply .pre-commit-config.yaml:56]: Nah, it's fine. Different strokes

**Pull request merged**